### PR TITLE
promise guard to return static fn

### PR DESCRIPTION
**Pull request by max-degterev**

### CODE REVIEW

File: src/use_promise_guard/index.es [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
       .then((result) => {
         if (isMounted.current) resolve(result);
       })
       .catch((result) => {
         if (isMounted.current) reject(result);
       });
-  });
-
-  return execute;
+  }), []);
```
Comment (peterhass) [inline src/use_promise_guard/index.es:15]: think you need `getPromise` as dependency in there otherwise you are working with a old `getPromise` function

**Review by peterhass — CHANGES_REQUESTED** [review]

File: src/use_promise_guard/index.es (hunk 1 above)
Comment (max-degterev) [reply src/use_promise_guard/index.es:15]: it's within the cached fn (argument)

Comment (peterhass) [reply src/use_promise_guard/index.es:15]: oh yeah. sorry, it's already late 💤

**Review by peterhass — APPROVED** [review]

**Comment by max-degterev** [comment]
I forgot why we didn't merge it, closing

File: src/use_promise_guard/index.es [hunk 2]
```diff
@@ -1,19 +1,18 @@
+import { useCallback } from 'react';
 import useMounted from '../use_mounted';
 
 const usePromiseGuard = () => {
   const isMounted = useMounted();
 
-  const execute = (getPromise) => new Promise((resolve, reject) => {
+  useCallback((getPromise) => new Promise((resolve, reject) => {
```
Comment (peterhass) [inline src/use_promise_guard/index.es:7]: Use [] for dependencies

**Pull request closed without merging**